### PR TITLE
TST: parametrize tests.indexing.test_float

### PR DESCRIPTION
**Pull request by jbrockmendel**
Working on making our exception-raising more consistent, getting these tests cleaned up will make that easier.

### CODE REVIEW

File: pandas/tests/indexing/test_floats.py [hunk 1]
```diff
@@ -70,97 +80,107 @@ def test_scalar_error(self, index_func):
             tm.makePeriodIndex,
         ],
     )
-    def test_scalar_non_numeric(self, index_func):
+    @pytest.mark.parametrize("klass", [Series, DataFrame])
```
Comment (WillAyd) [inline pandas/tests/indexing/test_floats.py]: Could be a logical follow up to make a `frame_or_series` fixture (we already have `index_or_series` in conftest) as used elsewhere for sure

File: pandas/tests/indexing/test_floats.py [hunk 2]
```diff
@@ -534,102 +541,94 @@ def test_slice_integer_frame_getitem(self, index_func):
...
+                "cannot do slice indexing "
+                r"on {klass} with these indexers \[0\.5\] of "
+                "type float".format(klass=type(index).__name__)
+            )
+            with pytest.raises(TypeError, match=msg):
+                s[l]
 
-        f(lambda x: x.loc)
+        # setitem
+        for l in [slice(3.0, 4), slice(3, 4.0), slice(3.0, 4.0)]:
```
Comment (WillAyd) [inline pandas/tests/indexing/test_floats.py]: Possible to parametrize this since done in other tests? Maybe need to split off?

**Review by WillAyd — COMMENTED** [review]
cool looks good. I know this will take a few passes so fine to merge this if it helps continue along

File: pandas/tests/indexing/test_floats.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/test_floats.py]: this will need to be split off in a couple of places, will do in a separate pass

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jbrockmendel** [comment]
@jreback gentle ping; lots of PRs are touching this file and it is particularly hard to rebase

File: pandas/tests/indexing/test_floats.py [hunk 3]
```diff
@@ -5,6 +5,16 @@
 import pandas._testing as tm
 
 
+def gen_obj(klass, index):
```
Comment (jreback) [inline pandas/tests/indexing/test_floats.py]: ok, should be turned into a fixture at some point (but understand for now)

**Pull request merged**